src/functions.php

```diff
@@ -101,8 +153,21 @@ function choose_handler(): callable
 
 /**
  * Get the default User-Agent string to use with Guzzle
+ *
+ * @deprecated
  */
 function default_user_agent(): string
+{
+    @trigger_error(sprintf('Function %s is deprecated and will be removed on guzzle 8.0', __FUNCTION__), E_USER_DEPRECATED);
```

Comment:
Not sure about this. Though it's kind of internal, but others can only read from it anyway, not modify. If we make the function private, there will be no way to just read the default agent.

I try to think about it as a constant, in this case IMO it would be a public constant like
```php
class Client
{
    public const DEFAULT_USER_AGENT = 'GuzzleHttp';
}
```

So for me it's a constant that is calculated at the first access (based on the environment). Thinking this way, I'm not sure why we should hide/protect it.